src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,33 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        if (rectangle.color == null || rectangle.width == null || rectangle.length == null
+                || color == null || width == null || length == null) {
+            return this.hashCode() == rectangle.hashCode();
+        }
+        return width.equals(rectangle.width) && length.equals(rectangle.length)
+                && color.equals(rectangle.color);
```

Comment:
see checklist: While comparing fields at equals, use single return statement for all fields(e.g. return check1 && check2 && check3;).

Reply:
fixed